### PR TITLE
Feature/sentry integration for dmsg server

### LINKED ISSUE #35: Optinal Sentry error loggining
**Feature description**

Sentry error logging for dmsg-server 

**Is your feature request related to a problem? Please describe.**
Our dmsg-server instances which we run sometimes restarts after panic and information from panic stack trace is not enough to identify error, so it will be nice to add additional logging that captures values of variables and sends them to Sentry. 

**Describe the solution you'd like**
I want to implement a command-line flag that accepts Sentry dsn and sends data on panic.

### PR DESCRIPTION
**Pull request by taras-skycoin**
Fixes #35 

 Changes:	
-	Add code for capturing panics and errors in Sentry.

How to test this PR:
Ask permission for https://sentry.io/organizations/none-9ua/issues/?project=5518253 and explore errors.

### CODE REVIEW

File: cmd/dmsg-server/commands/root.go [hunk 1]
```diff
@@ -53,6 +59,22 @@ var rootCmd = &cobra.Command{
 		if err := sf.ParseConfig(os.Args, true, &conf); err != nil {
 			log.WithError(err).Fatal()
 		}
+		if sentryDSN != "" {
+			err := sentry.Init(sentry.ClientOptions{
+				Dsn: sentryDSN,
+				// Either set environment and release here or set the SENTRY_ENVIRONMENT
```
Comment (nkryuchkov) [inline cmd/dmsg-server/commands/root.go]: What do you think about removing these comments and delete zero values from this structure?

File: cmd/dmsg-server/commands/root.go [hunk 2]
```diff
@@ -53,6 +59,22 @@ var rootCmd = &cobra.Command{
...
+				// and SENTRY_RELEASE environment variables.
+				Environment: "",
+				Release:     "",
+				// Enable printing of SDK debug messages.
+				// Useful when getting started or trying to figure something out.
+				Debug: true,
+			})
+			if err != nil {
+				log.Fatalf("sentry.Init: %s", err)
```
Comment (nkryuchkov) [inline cmd/dmsg-server/commands/root.go:67]: What is expected to happen if a Sentry server is not available? Should the program restart in a loop in this case?

File: cmd/dmsg-server/commands/root.go [hunk 3]
```diff
@@ -53,6 +59,22 @@ var rootCmd = &cobra.Command{
...
+			}
+			defer sentry.Flush(2 * time.Second)
```
Comment (nkryuchkov) [inline cmd/dmsg-server/commands/root.go:69]: Why is this value equal to 2 seconds?

File: server.go [hunk 4]
```diff
@@ -116,6 +117,7 @@ func (s *Server) Serve(lis net.Listener, addr string) error {
 
 	log.Info("Accepting sessions...")
 	s.readyOnce.Do(func() { close(s.ready) })
+	defer sentry.Recover()
```
Comment (nkryuchkov) [inline server.go:120]: If the `--sentry` flag is not passed and `sentry.Init` is not called, would this line work as expected?

Comment (taras-skycoin) [reply server.go:120]: It will work as before.

File: cmd/dmsg-server/commands/root.go (hunk 3 above)
Comment (taras-skycoin) [reply cmd/dmsg-server/commands/root.go:69]: I copied it from the official documentation. I think they do not allow to allow requests too often and have implemented bufferasation of events.

File: cmd/dmsg-server/commands/root.go (hunk 2 above)
Comment (taras-skycoin) [reply cmd/dmsg-server/commands/root.go:67]: I just checked this case with Sentry domain shadowed in `/etc/hosts` and everything works like no string passed at all, so everything is fine.

**Review by nkryuchkov — APPROVED** [review]

**Review by jdknives — APPROVED** [review]

**Pull request merged**